src/test/scala/com/advancedtelematic/campaigner/db/CampaignStatusRecalculateSpec.scala

```diff
@@ -0,0 +1,121 @@
+package com.advancedtelematic.campaigner.db
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import akka.stream.Materializer
+import akka.testkit.TestKit
+import cats.data.NonEmptyList
+import com.advancedtelematic.campaigner.data.DataType._
+import com.advancedtelematic.campaigner.data.Generators._
+import com.advancedtelematic.campaigner.util.CampaignerSpecUtil
+import com.advancedtelematic.campaigner.util.DatabaseUpdateSpecUtil
+import com.advancedtelematic.libats.test.DatabaseSpec
+import org.scalacheck.Gen
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import slick.jdbc.MySQLProfile.api._
+
+final class CampaignStatusRecalculateSpec
+    extends TestKit(ActorSystem("CampaignStatusRecalculateSpec"))
+    with AsyncFlatSpecLike
+    with ScalaFutures
+    with DatabaseSpec
+    with CampaignSupport
+    with GroupStatsSupport
+    with DeviceUpdateSupport
+    with UpdateSupport
+    with Matchers
+    with DatabaseUpdateSpecUtil
+    with CampaignerSpecUtil {
+
+  import SlickUtil._
+
+  implicit lazy val ec: ExecutionContext = system.dispatcher
+  implicit val mat: Materializer = ActorMaterializer()
+
+  it should "set campaign status to `cancelled` if a group was cancelled" in {
+    val groups = GroupWithDevices.genNonEmptyListOf.generate
+
+    for {
+      campaign <- createCampaignWithoutStatus(maybeGroups = Some(groups.map(_.id)))
+      _ <- insertGroupStatsFor(campaign.id, groups.toList)
+      _ <- groupStatsRepo.updateGroupStatsAction(
+        campaign.id, groups.head.id, GroupStatus.cancelled, Stats(0, 0)).run
+      _ <- new CampaignStatusRecalculate().run
+      updatedCampaign <- campaignRepo.find(campaign.id)
+    } yield {
```

Comment:
curly braces not needed for a single statement

Reply:
Same in all examples

Reply:
Sure. Do you follow a particular code style in projects? Where can I take a look at it? Would be nice to know the rules before playing the game :)

Reply:
We did at some point use scalastyle.xml but it caused more discussion than the time to discuss the changes ;) So we don't follow any particular code style except for the projects that do include scalastyle.xml

Reply:
Okay, got it. Thanks!